File: src/basic/_1014/1014.c
#include<stdio.h>
#include<string.h>
int main(){
    char days[7][4]={"MON","TUE","WED","THU","FRI","SAT","SUN"};
    char s1[80],s2[80],s3[80],s4[80];
    scanf("%s%s%s%s",s1,s2,s3,s4);
    int week = -1;
    int hour = -1;
    int count = 0;

    int len12 = strlen(s1) > strlen(s2) ? strlen(s2) : strlen(s1);
    for (int i = 0; i < len12; i++) {
        if (s1[i] == s2[i]) {
            if (count == 0) {
                if (s1[i] >= 'A' && s1[i] <= 'G') {
                    week = s1[i] - 'A';
                    count++;
                }
            } else {
                if (s1[i] >= 'A' && s1[i] <= 'N') {
                    hour = s1[i] - 'A' + 10;
                    break;
                } else if (s1[i] >= '0' && s1[i] <= '9') {
                    hour = s1[i] - '0';
                    break;
                }
            }
        }
    }

    int min = -1;
    int len34 = strlen(s3) > strlen(s4)? strlen(s4) : strlen(s3);
    for (int i = 0; i < len34; i++) {
        if (s3[i] == s4[i] && (s3[i] >= 'a' && s3[i] <= 'z' || s3[i] >= 'A' && s3[i] <= 'Z')) {
            min = i;
            break;
        }
    }
    printf("%s %02d:%02d\n",days[week],hour,min);
    return 0;
}

File: src/basic/_1014/1014.js
var readline = require('readline')

rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})
var inputArr = []
rl.on('line', function(input) {
  inputArr.push(input)
  var str1 = inputArr[0]
  var str2 = inputArr[1]
  var str3 = inputArr[2]
  var str4 = inputArr[3]
  if (inputArr.length === 4) {
    var reslut = deal(str1,str2,str3,str4)
    console.log(reslut);
    rl.close()
  }
})
function deal(str1,str2,str3,str4) {
  var week = ''
  var hours = ''
  var minute = ''
  for(var i = 0 ; i < Math.min(str1.length,str2.length); i++) {
    if (str1.charAt(i) === str2.charAt(i) && /[A-G]/.test(str1.charAt(i))) {
      week = str1.charAt(i)
      var j = i
      break
    }
  }
  for(var i = j+1 ; i < Math.min(str1.length,str2.length); i++) {
    if (str1.charAt(i) === str2.charAt(i) && /[A-N0-9]/.test(str1.charAt(i))) {
      hours = str1.charAt(i)
      break
    }
  }
  for(var i = 0 ; i < Math.min(str3.length,str4.length); i++) {
    if (str3.charAt(i) === str4.charAt(i) && /[a-zA-Z]/.test(str3.charAt(i))) {
      minute = i
      break
    }
  }
  var weekArr1 = ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN']
  week = weekArr1[(parseInt((week+''), 17) + '').split('')[1]]
  hours = (parseInt(hours, 24) < 10 ?('0' + parseInt(hours, 24)):parseInt(hours, 24))
  minute = (minute < 10 ?('0' + minute):minute)
  return week+ ' '+hours + ':'+minute
}
